packages/swc-configs/src/build.ts

```diff
@@ -0,0 +1,55 @@
+import type { Config } from "@swc/core";
+import { resolveOverrides, type ConfigOverride } from "./resolveOverrides.ts";
+
+export const DefaultBuildConfig = {
+    jsc: {
+        parser: {
+            syntax: "typescript",
+            tsx: true
+        },
+        // The output environment that the code will be compiled for.
+        target: "es2022",
+        // View https://swc.rs/docs/configuration/minification for options.
+        minify: {
+            compress: true,
+            mangle: true
+        },
+        transform: {
+            react: {
+                // Use "react/jsx-runtime".
+                runtime: "automatic",
+                // Use the native "Object.assign()" instead of "_extends".
+                useBuiltins: true
+            }
+        },
+        // Import shims from an external module rather than inlining them in bundle files to greatly reduce the bundles size.
+        // Requires to add "@swc/helpers" as a project dependency
+        externalHelpers: true
+    },
+    module: {
+        // The output module resolution system that the code will be compiled for.
+        type: "es6",
+        // Prevent SWC from exporting the `__esModule` property.
+        strict: true,
+        // Preserve dynamic imports.
+        ignoreDynamic: true
+    }
+};
+
+export interface DefineBuildConfigOptions {
+    parser?: "ecmascript";
+    configOverride?: ConfigOverride;
+}
+
+export function defineBuildConfig({ parser, configOverride }: DefineBuildConfigOptions = {}) {
+    const config = { ...DefaultBuildConfig } as Config;
+
+    if (parser === "ecmascript") {
+        config.jsc!.parser = {
```

Comment:
Depending on how you fix the previous comment, this is a great opportunity to use the satisfies operator here. 

```
 export const DefaultBuildConfig = {
... all the props
} satisfies Config;


```
The satisfies operator ensures the object fits Config but you shouldn't have to specify the config.jsc! since typescript knows it's been defined in the Defaultconfig object

Reply:
Handle this one in DM, will not change anything.